### PR TITLE
DUPLO-15256 Added Snapshot attribute support for ecache

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Added support for attribute `snapshot_name`, `snapshot_arns`, `snapshot_retention_limit` for `duplocloud_ecache_instance`
## Summary of changes

This PR does the following:

- Added snapshot attribute snapshot_name, snapshot_arns (conflicts with snapshot_name) and snapshot_retention_limit 
- Updated docs

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [✓ ] Manually, on a remote test system

## Describe any breaking changes

- ...

___

## **Type**
enhancement

___

## **Description**
- Added support for managing snapshot attributes in ecache instances, including `snapshot_name`, `snapshot_arn`, and `snapshot_retention_limit`.
- Updated the Go structs and Terraform schema to support these new attributes.
- Provided documentation for the new snapshot attributes in the ecache instance resource.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_ecache_instance.go</strong><dd><code>Add Snapshot Attributes to ECache Instance Schema</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/resource_duplo_ecache_instance.go

<li>Added support for snapshot attributes in the ecache instance schema.<br> <li> Updated the expand and flatten functions to handle new snapshot <br>attributes.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/518/files#diff-0d4e960f6d6fea601a3f043c811148ed9f58959f772948285f4a11cc38708a0c">+40/-14</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ecache_instance.go</strong><dd><code>Update DuploEcacheInstance Struct for Snapshot Support</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplosdk/ecache_instance.go

<li>Added new fields to the DuploEcacheInstance struct to support snapshot <br>attributes.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/518/files#diff-2824ae62bf701d8827c13dc9a091306010c3bfae2f0a77fce94e0981b994f9b1">+18/-15</a>&nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>ecache_instance.md</strong><dd><code>Document Snapshot Attributes in ECache Instance</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
docs/resources/ecache_instance.md

<li>Documented the new snapshot attributes in the ecache instance resource <br>documentation.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/518/files#diff-043ac248497c05f1e877b1714946cf707b361bbd70c813e754295d7750a58398">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-15256 TF: duplocloud_ecache_instance: Add support to create redis with snapshot and `Snapshot Retention Limit` field](https://app.clickup.com/t/8655600/DUPLO-15256)

**Comment by sp1676** [comment]
/'update_changelog --pr_update_changelog.push_changelog_changes=true'

File: duplocloud/resource_duplo_ecache_instance.go [hunk 1]
```diff
@@ -157,6 +157,26 @@ func ecacheInstanceSchema() map[string]*schema.Schema {
 			Computed:         true,
 			ValidateFunc:     validation.IntBetween(1, 500),
 		},
+		"snapshot_arn": {
+			Description:   "Specify the ARN of a Redis RDB snapshot file stored in Amazon S3.",
+			Type:          schema.TypeString,
+			Optional:      true,
+			Computed:      true,
+			ConflictsWith: []string{"snapshot_name"},
```
Comment (sp1676) [inline duplocloud/resource_duplo_ecache_instance.go:165]: Can you check the comment from codium about snapshot_name also needing ConflictsWith?

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_ecache_instance.go:165]: I am still working on this closing this PR will raise new one

**Pull request closed without merging**